replica/diff_disk.go

```diff
@@ -166,7 +163,7 @@ func (d *diffDisk) fullWriteAt(buf []byte, offset int64) (int, error) {
 		if val := d.location[startSector+i]; val == 0 {
 			d.UsedLogicalBlocks++
 			d.UsedBlocks++
-		} else if val != target {
+		} else if val != target && d.preloadStatus == types.Done {
```

Comment:
you need to add a comment here, and also, at the end of the 'for' loop where sendToCreateHole is called